mesonbuild/linkers/detect.py

```diff
@@ -151,7 +151,7 @@ def guess_nix_linker(env: 'Environment', compiler: T.List[str], comp_class: T.Ty
     _, o, e = Popen_safe(compiler + check_args)
     v = search_version(o + e)
     linker: DynamicLinker
-    if o.startswith('LLD'):
+    if re.search(r'^.*\bLLD\b', o):
```

Comment:
`partition` also takes time and memory proportional to the size of the string. Sure `--help` is not going to be 1G large, but it seems just wrong if we know the first line is the only one that is involved; same for `'LLD' in o`.

Reply:
Maybe it seems wrong, but I don't think it actually is. A really smart search algorithm will figure out what it's searching and then select the methodology that is best suited to the input. In this case it might either check the string size and switch to regex after a certain cutoff point, or simply assume that program --help text can't be megabytes in size without it being broken in unusually bizarre ways.